### PR TITLE
IBP-5408 pre populate export data and labels with selected columns and entry details

### PR DESCRIPTION
**Pull request by cuenyad**
Please, Review.

Included:
https://github.com/IntegratedBreedingPlatform/Workbench/pull/653

### CODE REVIEW

File: src/main/java/org/ibp/api/rest/labelprinting/LotLabelPrinting.java [hunk 1]
```diff
@@ -187,6 +188,11 @@ void validateLabelsGeneratorInputData(final LabelsGeneratorInput labelsGenerator
 		super.validateLabelsGeneratorInputData(labelsGeneratorInput, programUUID);
 	}
 
+	@Override
+	public LabelPrintingPresetDTO getDefaultSetting(final LabelsInfoInput labelsInfoInput, final String programUUID) {
+		return null;
+	}
```
Comment (nahuel-soldevilla) [inline src/main/java/org/ibp/api/rest/labelprinting/LotLabelPrinting.java:194]: We can return Optional but it'll be more or less the same, as we need to return empty to the frontend

File: src/main/java/org/ibp/api/rest/labelprinting/GermplasmListLabelPrinting.java [hunk 2]
```diff
@@ -286,4 +292,91 @@ private void getEntryDetailFieldDataRowValue(
...
+
+		});
+
+		final LabelPrintingPresetDTO labelPrintingPresetDTO = new LabelPrintingPresetDTO();
+		final FilePresetConfigurationDTO filePresetConfigurationDTO = new FilePresetConfigurationDTO();
+		filePresetConfigurationDTO.setOutputType(FileType.XLS.getExtension());
+		labelPrintingPresetDTO.setFileConfiguration(filePresetConfigurationDTO);
+		labelPrintingPresetDTO.setSelectedFields(Arrays.asList(fields));
+		labelPrintingPresetDTO.setBarcodeSetting(new LabelPrintingPresetDTO.BarcodeSetting(false, false, null));
+		labelPrintingPresetDTO.setToolId(23);
```
Comment (nahuel-soldevilla) [inline src/main/java/org/ibp/api/rest/labelprinting/GermplasmListLabelPrinting.java:376]: we've been using the fieldbook toolid, we'll need to review that at some point, not much use probably

File: src/main/java/org/ibp/api/rest/labelprinting/GermplasmListLabelPrinting.java [hunk 3]
```diff
@@ -286,4 +292,91 @@ private void getEntryDetailFieldDataRowValue(
...
+					fields.add(TermId.CROSS_MALE_GID.getId());
+					break;
+				case MALE_PARENT:
+					fields.add(TermId.CROSS_MALE_PREFERRED_NAME.getId());
+					break;
+				case FEMALE_PARENT:
+					fields.add(TermId.CROSS_FEMALE_PREFERRED_NAME.getId());
+					break;
+				default:
+					if (dto.getTermId() == GermplasmListLabelPrinting.DRVNM_ID || dto.getTermId() > 10000) {
```
Comment (nahuel-soldevilla) [inline src/main/java/org/ibp/api/rest/labelprinting/GermplasmListLabelPrinting.java:360]: Change GermplasmLabelPrinting#MAX_FIXED_TYPE_INDEX to protected and use that instead

**Review by nahuel-soldevilla — APPROVED** [review]
No more concerns from me, looks fine

File: src/main/java/org/ibp/api/rest/labelprinting/GermplasmListLabelPrinting.java (hunk 2 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/rest/labelprinting/GermplasmListLabelPrinting.java:376]: Ok, I will remove it.

**Review by clarysabel — APPROVED** [review]

**Pull request merged**